sky/serve/replica_managers.py

```diff
@@ -581,37 +678,88 @@ def _download_and_stream_logs(info: ReplicaInfo):
                                                     replica_id)
         assert info is not None
 
-        if sync_down_logs:
+        is_launching = False
+        if replica_id in self._launch_process_pool:
+            is_launching = True
+            launch_process = self._launch_process_pool[replica_id]
+            if launch_process.is_alive():
+                logger.info(f'Terminating launch process for replica '
+                            f'{replica_id}...')
+                launch_process.terminate()
+                launch_process.join()
+            del self._launch_process_pool[replica_id]
+
+        if sync_down_logs and not is_launching:
             _download_and_stream_logs(info)
 
         logger.info(f'preempted: {info.status_property.preempted}, '
                     f'replica_id: {replica_id}')
         log_file_name = serve_utils.generate_replica_down_log_file_name(
             self._service_name, replica_id)
+        max_retry = 3
```

Comment:
we could swap the position of `max_retry` and `delay` and don't specify this argument